### DISCUSSION TITLE
Shield for the date of the last github action / workflow

### DISCUSSION THREAD

**Discussion by tsuza** [badges/shields#10081] [Q&A]
I've found a shield that shows *whether* an action has passed, but nothing related to the **date** of the latest github action. I was wondering if it was available. I wasn't able to find anything, neither in the shields.io website nor in the github endpoint list.

**Comment by calebcartwright** [comment]
Are you essentially asking for a "last invoked" badge for github actions?

**Reply by tsuza** [reply]
Yeah, you're correct.

**Comment by Zoobdude** [comment]
I'd also like this functionality. Going to see if I can build it myself.

**Reply by tsuza** [reply]
Nice, glad to see that I'm not the only one to think that!

Let me know if you get it working. I was looking for some endpoints, but I wasn't able to find any ( I might have missed it ).

**Reply by Zoobdude** [reply]
It looks like [list workflow runs for a workflow](https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28#list-workflow-runs-for-a-workflow) (just set `per_page` to 1 to get the most recent run) provides the run date and time.

**Comment by chris48s** [comment]
There is some context and history to be aware of when it comes to trying to get information about github workflow runs from the api, which I collected here: https://github.com/badges/shields/issues/9189#issuecomment-1556265808
Last run of a workflow _against any branch/ref_ should be fine (I think :crossed_fingers: ).
Last run of a workflow _on the default branch_ or _against a specific ref_ will have all the same problems we hit there.
